packages/core-contracts/contracts/utils/ERC20.sol

```diff
@@ -0,0 +1,93 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import "../interfaces/IERC20.sol";
+
+abstract contract ERC20 is IERC20 {
+    string public override name;
+
+    string public override symbol;
+
+    uint8 public immutable override decimals;
+
+    mapping(address => uint256) public override balanceOf;
```

Comment:
Doesn't this collide with the `balanceOf` function?

Reply:
nope, solc 0.7 I think allows public vars to override functions. The getter will be produced no matter what.